src/Simplex/FileTransfer/Agent.hs

```diff
@@ -288,12 +288,16 @@ runXFTPRcvLocalWorker c Worker {doWork} = do
 
 xftpDeleteRcvFile' :: AgentMonad m => AgentClient -> RcvFileId -> m ()
 xftpDeleteRcvFile' c rcvFileEntityId = do
-  RcvFile {rcvFileId, prefixPath, status} <- withStore c $ \db -> getRcvFileByEntityId db rcvFileEntityId
-  if status == RFSComplete || status == RFSError
-    then do
-      removePath prefixPath
-      withStore' c (`deleteRcvFile'` rcvFileId)
-    else withStore' c (`updateRcvFileDeleted` rcvFileId)
+  rcvFile <- withStore c $ \db -> getRcvFileByEntityId db rcvFileEntityId
+  handleError (const $ pure ()) $ withStore' c (`getRcvFileEntityRedirect` rcvFileEntityId) >>= mapM_ remove
```

Comment:
maybe better to search by db id which is already known in rcvFile? it's a primary key